### ISSUE TITLE
dot_product_attention triggers FloatingPointError in convert_element_type with float64 mask and debug_infs=True

### ISSUE THREAD

**Issue by ALinrunrun** [jax-ml/jax#37422]
### Description

`jax.nn.dot_product_attention` triggers a `FloatingPointError` when `jax.debug_infs(True)` is enabled.

The inputs are all zeros with `float64` dtype, and the mask is a simple boolean mask. The expected attention output should be finite, but JAX reports an invalid `inf` value in `convert_element_type`.

Minimal reproducible example:

```
import sys

try:
    import jax
    import jax.numpy as jnp
    import numpy as np
except ImportError as e:
    print(f"missing dep: {e}")
    sys.exit(2)

def main() -> int:
    T, S, K, N, H = 5, 7, 3, 6, 11
    mask = np.r_[:S] <= S // 2

    try:
        jax.config.update("jax_enable_x64", True)
        with jax.debug_infs(True):
            out = jax.nn.dot_product_attention(
                np.zeros((T, N, H), dtype=np.float64),
                np.zeros((S, K, H), dtype=np.float64),
                np.zeros((S, K, H), dtype=np.float64),
                mask=mask,
            )
            out.block_until_ready()
        print("NOT REPRODUCED: no FloatingPointError with debug_infs=True")
        return 1
    except FloatingPointError as e:
        print(f"BUG REPRODUCED: FloatingPointError triggered: {e}")
        return 0
    except Exception as e:
        print(f"NOT REPRODUCED: different exception {type(e).__name__}: {e}")
        return 1

if __name__ == "__main__":
    raise SystemExit(main())
```

### System info (python version, jaxlib version, accelerator, etc.)

```
WARNING:2026-05-07 00:33:38,915:jax._src.xla_bridge:794: An NVIDIA GPU may be present on this machine, but a CUDA-enabled jaxlib is not installed. Falling back to cpu.
jax:    0.6.2
jaxlib: 0.6.2
numpy:  2.2.6
python: 3.10.20 (main, Mar 11 2026, 17:46:40) [GCC 14.3.0]
device info: cpu-1, 1 local devices
process_count: 1
platform: uname_result(system='Linux', node='MSI', release='5.15.167.4-microsoft-standard-WSL2', version='#1 SMP Tue Nov 5 00:21:55 UTC 2024', machine='x86_64')

nvidia-smi:
NVIDIA-SMI 545.23.06
Driver Version: 545.92
CUDA Version: 12.3
GPU: NVIDIA GeForce RTX 3080, 8192 MiB

Note: Reproduced on CPU fallback because CUDA-enabled jaxlib is not installed.
```

**Comment by ShresthSamyak** [comment]
Fix in #37457
